flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSink.java

```diff
@@ -74,10 +76,13 @@
 	public HiveTableSink(JobConf jobConf, ObjectPath tablePath, CatalogTable table) {
 		this.jobConf = jobConf;
 		this.tablePath = tablePath;
-		this.catalogTable = table;
+
+		Preconditions.checkArgument(table instanceof CatalogTableImpl);
+		this.catalogTable = HiveTableUtil.toHiveCatalogTable(table);
```

Comment:
It seems that all we  need is to change some field types to their bridged ones. This can be done by building a new TableSchema from the original TableSchema. In other word, we don't need to create a new CatalogTable instance.

Reply:
You're right. But `CatalogTableImpl` currently does not have setSchema method. Should I add one?